jet_images_ml.py

```diff
@@ -116,170 +72,202 @@ def flatten_layer(layer):
+fc1 = fc_layer(input=layer_flat,num_inputs=num_features,num_outputs=fc_size, name="fc1")
 sigmoid = tf.nn.sigmoid(fc1)
 tf.summary.histogram("fc1/sigmoid", sigmoid)
-fc2 = fc_layer(input=sigmoid,num_inputs=fc_size,num_outputs=num_classes,name="fc2")
+fc2 = fc_layer(input=sigmoid,num_inputs=fc_size,num_outputs=num_classes, name="fc2")
 
 y_pred = fc2
 y_pred_cls = tf.argmax(y_pred, dimension=1)
 
 
+"""
+Learning configuration
+"""
+
+# batch size
+batch_size = 25
+# validation split
+validation_size = .2
+learning_rate = 0.0005
+
+# how long to wait after validation loss stops improving before terminating training
+early_stopping = None  # use None if you don't want to implement early stopping
+
+
+"""
+Data set Configuration
+"""
+
+train_path = 'data/jet_images_sample/train'
+test_path = 'data/jet_images_sample/test'
+
+data = dataset.read_train_sets(train_path, img_size, classes, validation_size=validation_size)
+test_images, test_ids, labels = dataset.read_test_set(test_path, img_size, classes)
+
+print("Size of:")
+print("- Training-set:\t\t{}".format(len(data.train.labels)))
+print("- Test-set:\t\t{}".format(len(test_images)))
+print("- Validation-set:\t{}".format(len(data.valid.labels)))
+
+num_batches = int(data.train.num_examples/batch_size)
+
+"""
+Loss Function + Optimization
+"""
+
 with tf.name_scope("cost"):
-    cross_entropy = tf.nn.sigmoid_cross_entropy_with_logits(logits=fc2,labels=y_true)
+    cross_entropy = tf.nn.sigmoid_cross_entropy_with_logits(logits=fc2, labels=y_true)
     cost = tf.reduce_mean(cross_entropy)
-    tf.summary.scalar("cost",cost)
+    tf.summary.scalar("cost", cost)
 with tf.name_scope("Optimize"):
-    optimizer = tf.train.AdamOptimizer(learning_rate=0.0005).minimize(cost)
+    optimizer = tf.train.AdamOptimizer(learning_rate=learning_rate).minimize(cost)
     
 with tf.name_scope("accuracy"):
     correct_prediction = tf.equal(y_pred_cls, y_true_cls)
     accuracy = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))
-    tf.summary.scalar("accuracy",accuracy)
-
-
+    tf.summary.scalar("accuracy", accuracy)
 
-
-train_batch_size = batch_size
-merged_summary=tf.summary.merge_all()
 session.run(tf.global_variables_initializer())
-writer= tf.summary.FileWriter('/tmp/classifier/10')
-writer.add_graph(session.graph)
-
-
 
-def print_progress(epoch, feed_dict_train, feed_dict_validate, val_loss):
-    # Calculate the accuracy on the training-set.
-    acc = session.run(accuracy, feed_dict=feed_dict_train)
-    val_acc = session.run(accuracy, feed_dict=feed_dict_validate)
-    msg = "Epoch {0} --- Training Accuracy: {1:>6.1%}, Validation Accuracy: {2:>6.1%}, Validation Loss: {3:.3f}"
-    print(msg.format(epoch + 1, acc, val_acc, val_loss))
+logdir = 'logs/classifier/10'
+writer = tf.summary.FileWriter(logdir)
+writer.add_graph(session.graph)
 
-    
-    
-    
+eval_writer = tf.summary.FileWriter(logdir + '_eval')
 
+x_batch, _, _, _ = data.train.next_batch(batch_size)
```

Comment:
Can you please tell me why are we adding this line?  When I removed this in the large dataset the accuracy is increasing.